### PR TITLE
V1 of sending username to blazar for manual build triggers

### PR DESCRIPTION
**Pull request by jonathanwgoodwin**
@zdhickman @jhaber @gchomatas 
LMK what you think

### CODE REVIEW

File: BlazarService/src/main/java/com/hubspot/blazar/resources/RepositoryBuildResource.java [hunk 1]
```diff
@@ -41,20 +42,23 @@ public RepositoryBuildResource(RepositoryBuildService repositoryBuildService,
...
     }
-
-    // TODO capture user
-    BuildTrigger buildTrigger = BuildTrigger.forUser("unknown");
+    BuildTrigger buildTrigger;
+    if (username.isPresent()) {
+      buildTrigger = BuildTrigger.forUser(username.get());
+    }  else {
+      buildTrigger = BuildTrigger.forUser("unknown");
+    }
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/resources/RepositoryBuildResource.java]: You can do `BuildTrigger buildTrigger = BuildTrigger.forUser(username.or("unknown"));`

File: BlazarUI/app/scripts/data/BranchApi.js [hunk 2]
```diff
@@ -77,9 +78,9 @@ function triggerBuild(params, moduleIds, downstreamModules, resetCache, cb) {
   if (moduleIds === null) {
     moduleIds = [];
   }
-
+  const username = Cookies.get(config['usernameCookie']) || null;
   const buildPromise = new Resource({
-    url: `${config.apiRoot}/branches/builds/branch/${params.branchId}`,
+    url: `${config.apiRoot}/branches/builds/branch/${params.branchId}?username=${username}`,
```
Comment (zdhickman) [inline BlazarUI/app/scripts/data/BranchApi.js]: Is the behavior for Jackson that if you send a query param of the type `Optional` with a value of `null`, it's bound as `Optional.absent()`?

**Comment by zdhickman** [comment]
Anyway otherwise, lgtm

File: BlazarUI/app/scripts/data/BranchApi.js [hunk 3]
```diff
@@ -77,9 +78,9 @@ function triggerBuild(params, moduleIds, downstreamModules, resetCache, cb) {
   if (moduleIds === null) {
     moduleIds = [];
   }
-
+  const username = Cookies.get(config['usernameCookie']) || null;
```
Comment (zdhickman) [inline BlazarUI/app/scripts/data/BranchApi.js]:
```
const username = config.usernameCookie && Cookies.get(config.usernameCookie) ? `username=${Cookies.get(config.usernameCookie)}` : '';
const buildPromise = new Resource({
  url: `${config.apiRoot}/branches/builds/branch/${params.branchId}?${username}`
```

**Pull request merged**